src/coreclr/vm/codeman.h

```diff
@@ -287,6 +296,13 @@ struct CodeHeader
         SUPPORTS_DAC;
         return pRealCodeHeader->nUnwindInfos;
     }
+
+    IsFuncletCache ComputeInitialIsFuncletCacheValue(PCODE codePtr)
+    {
+        if (GetNumberOfUnwindInfos() == 1)
+            return IsFuncletCache::IsNotFunclet; // Optimize the scenario where there are not any funclets in a method
+        return IsFuncletCache::NotSet;
+    }
```

Comment:
```suggestion
        bool MayHaveFunclets()
        {
            return GetNumberOfUnwindInfos() > 1;
        }
```
It would be nice if IsFuncletCache is kept as a private detail of EECodeInfo.